pkg/workflows/builder_test.go

```diff
@@ -0,0 +1,296 @@
+package workflows_test
+
+import (
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/yaml"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/targets/chainwriter"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers/streams"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/testdata/fixtures/capabilities/notstreams"
+)
+
+// Note that the set of tests in this file cover the conversion from existing YAM -> code
+// along with testing the structure of what is generated from the builders.
+// This implicitly tests the code generators functionally, as the generated code is used in the tests.
+
+type Config struct {
+	Workflow    workflows.NewWorkflowParams
+	Streams     *streams.TriggerConfig
+	Ocr         *ocr3.ConsensusConfig
+	ChainWriter *chainwriter.TargetConfig
+	TargetChain string
+}
+
+func NewWorkflowSpec(rawConfig []byte) (workflows.WorkflowSpec, error) {
+	conf := Config{}
+	if err := yaml.Unmarshal(rawConfig, &conf); err != nil {
+		return workflows.WorkflowSpec{}, err
+	}
+
+	workflow := workflows.NewWorkflow(conf.Workflow)
+	streamsTrigger := conf.Streams.New(workflow)
+
+	ocrInput := ocr3.ConsensusInput{Observations: workflows.ListOf[streams.Feed](streamsTrigger)}
+	consensus := conf.Ocr.New(workflow, "ccip_feeds", ocrInput)
+
+	input := chainwriter.TargetInput{SignedReport: consensus}
+	conf.ChainWriter.New(workflow, conf.TargetChain, input)
```

Comment:
Some users might prefer this style of omitting the intermediate vars :shrug: 
```suggestion
	consensus := conf.Ocr.New(workflow, "ccip_feeds", ocr3.ConsensusInput{
		Observations: workflows.ListOf[streams.Feed](streamsTrigger),
	})

	conf.ChainWriter.New(workflow, conf.TargetChain, chainwriter.TargetInput{
		SignedReport: consensus,
	})
```